app/src/main/java/com/example/neighborly/FeedFragment.java

```diff
@@ -72,29 +77,62 @@ public void onClick(View v) {
+        requestMessageEditor.setText(String.format(getString(R.string.neighborly_send_help_message), itemName));
         TextView intro = popupRequestDialog.findViewById(R.id.intro);
         intro.setText(String.format(introText, UserModelDataHolder.getInstance().getCurrentUser().getPresentedName()));
         ListView neighborsListView = popupRequestDialog.findViewById(R.id.foundNeighbors);
-        neighborsListView.setAdapter(new SearchResultListAdapter(getActivity(), foundItems, getContext()));
+        neighborsListView.setAdapter(new SearchResultListAdapter(getActivity(), foundItems, getContext(), itemName));
 
         sendButton = popupRequestDialog.findViewById(R.id.send);
         sendButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                Intent intent = new Intent(getActivity(), RequestActivity.class);
-                // todo get real id
-                //intent.putExtra("context", RequestActivity.itemRequestContext);
-                intent.putExtra("itemId", 0);
-                intent.putExtra("itemName", searchText.getText().toString());
-                startActivity(intent);
-                popupRequestDialog.dismiss();
+                EditText requestMessageEditor = popupRequestDialog.findViewById(R.id.editRequestMessage);
+                String newItemRequestContent = requestMessageEditor.getText().toString();
+                UserModel curUser = UserModelDataHolder.getInstance().getCurrentUser();
+
+                DatabaseReference requestsRef = database.getReference().child(Constants.REQUESTS);
+
+                // Generate a reference to a new location and add some data using push()
+                DatabaseReference requestKeyRef = requestsRef.push();
+
+                // Get the unique ID generated by push() by accessing its key
+                String requestId = requestKeyRef.getKey();
+                if (requestId != null) {
+                    RequestModel newRequest = new RequestModel(requestId, false, curUser.getId(),
+                            newItemRequestContent, itemName);
+                    Map<String, Object> requests = new HashMap<>();
+                    requests.put(requestId, newRequest);
+
+                    requestKeyRef.setValue(newRequest);
+                    addItemsUnderBuildingInDB(newRequest);
+
+                    Intent intent = new Intent(getActivity(), RequestActivity.class);
+                    intent.putExtra("requestType", RequestActivity.REQUEST_ITEM);
+                    intent.putExtra("requestId", requestId);
+                    startActivity(intent);
+
+                    popupRequestDialog.dismiss();
+                }
             }
         });
         popupRequestDialog.getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
         popupRequestDialog.show();
     }
+
+    private void addItemsUnderBuildingInDB(RequestModel newRequest) {
```

Comment:
should be "addRequestsUnderBuildingInDB"

Reply:
yup :) good notice. Will change it in next pull.